File: src/main/java/com/bot/commands/voice/VolumeCommand.java
package com.bot.commands.voice;

import com.bot.commands.VoiceCommand;
import com.bot.voice.VoiceSendHandler;
import com.jagrosh.jdautilities.command.CommandEvent;

public class VolumeCommand extends VoiceCommand {

	public VolumeCommand() {
		this.name = "volume";
		this.arguments = "<Volume 1-200>";
		this.help = "Sets the players volume";
	}

	@Override
	protected void executeCommand(CommandEvent commandEvent) {
		VoiceSendHandler handler = (VoiceSendHandler) commandEvent.getGuild().getAudioManager().getSendingHandler();
		int newVolume;
		try {
			if (handler == null) {
				commandEvent.replyWarning("I am not connected to a voice channel.");
				return;
			}

			newVolume = Integer.parseInt(commandEvent.getArgs().split(" ")[0]);
			if (newVolume > 200 || newVolume < 0) {
				throw new NumberFormatException();
			}
			if (handler.isLocked()) {
				commandEvent.replyWarning("Volume is currently locked. You need to unlock it to edit it.");
				return;
			}
			handler.getPlayer().setVolume(newVolume);
			commandEvent.reactSuccess();
		}
		catch (NumberFormatException e) {
			commandEvent.replyWarning("You can enter a volume between 0 and 200 to set.\nCurrent volume: " + handler.getPlayer().getVolume());
		}
	}
}

File: src/main/java/com/bot/commands/voice/VolumeCommand.kt
package com.bot.commands.voice

import com.bot.commands.VoiceCommand
import com.jagrosh.jdautilities.command.CommandEvent
import datadog.trace.api.Trace

class VolumeCommand : VoiceCommand() {
    init {
        name = "volume"
        arguments = "<Volume 1-200>"
        help = "Sets the players volume"
    }

    @Trace(operationName = "executeCommand", resourceName = "Volume")
    override fun executeCommand(commandEvent: CommandEvent) {
        val voiceConnection = guildVoiceProvider.getGuildVoiceConnection(commandEvent.guild)
        val newVolume: Int
        try {
            if (!voiceConnection.isConnected()) {
                commandEvent.replyWarning("I am not connected to a voice channel.")
                return
            }
            newVolume = commandEvent.args.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()[0].toInt()
            voiceConnection.setVolume(newVolume)
            commandEvent.reactSuccess()
        } catch (e: NumberFormatException) {
            commandEvent.replyWarning(
                """
    You can enter a volume between 0 and 200 to set.
    Current volume: ${voiceConnection.getVolume()}
    """.trimIndent()
            )
        }
    }
}